player/lua/select.lua

```diff
@@ -261,54 +261,92 @@ mp.add_key_binding(nil, "select-edition", function ()
     })
 end)
 
-mp.add_key_binding(nil, "select-subtitle-line", function ()
-    local sub = mp.get_property_native("current-tracks/sub")
+local function select_subtitle_line(data, codec)
+    local sub_lines = {}
+    local sub_times = {}
+    local default_item
+    local delay = mp.get_property_native("sub-delay")
+    local time_pos = mp.get_property_native("time-pos") - delay
+    local duration = mp.get_property_native("duration", math.huge)
+    local sub_content = {}
+
+    -- Strip HTML and ASS tags and process subtitles
+    for line in data:gmatch("[^\n]+") do
+        -- Clean up tags
+        local sub_line = line:gsub("<.->", "")         -- Strip HTML tags
+                             :gsub("{[\\=].-}", "")    -- Remove ASS formatting
+                             :gsub(".-]", "", 1)       -- Remove time info prefix
+                             :gsub("^m[l%s%-%d]+", "") -- Remove graphics code
+                             :gsub("\\h+", " ")        -- Replace '\h' tag
+
+        if codec == "subrip" or (sub_line ~= "" and sub_line:match("^%s+$") == nil) then
+            local sub_time = line:match("%d+") * 60 + line:match(":([%d%.]*)")
+            local time_seconds = math.floor(sub_time)
+
+            -- Use a dictionary to store the timestamp and content
+            if not sub_content[time_seconds] then
+                -- Initialize content list for that timestamp
+                sub_content[time_seconds] = {}
+            end
 
-    if sub == nil then
-        show_warning("No subtitle is loaded.")
-        return
-    end
+            -- If the subtitle content is not already in the list for that timestamp, add it
+            local content_exists = false
+            for _, exist_content in ipairs(sub_content[time_seconds]) do
```

Comment:
`exist_content` is incorrect English, this can be just `content`

Reply:
Done.